pdfbox/src/test/java/org/apache/pdfbox/pdmodel/font/TestFontEmbedding.java

```diff
@@ -371,4 +376,116 @@ void testReuseEmbeddedSubsettedFont() throws IOException
             assertEquals(text1 + " " + text2, extractedText.trim());
         }
     }
+
+    private class TrueTypeEmbedderTester extends TrueTypeEmbedder {
+        /**
+         * Common functionality for testing the TrueTypeFontEmbedder
+         *
+         */
+        TrueTypeEmbedderTester(PDDocument document, COSDictionary dict, TrueTypeFont ttf, boolean embedSubset) throws IOException {
+            super(document, dict, ttf, embedSubset);
+        }
+
+        @Override
+        protected void buildSubset(InputStream ttfSubset, String tag, Map<Integer, Integer> gidToCid) throws IOException {
+            // no-op.  Need to define method to extend abstract class, but
+            // this method is not currently needed for testing
+        }
+    }
+
+    /**
+     * Test that we validate embedding permissions properly for all legal permissions combinations
+     *
+     * @throws IOException
+     */
+    @Test
+    void testIsEmbeddingPermittedMultipleVersons() throws IOException
```

Comment:
This test is compatible with the version 3.0.x codebase.  However, it looks like the testing frameworks have changed significantly between version 2.0.x and 3.0.x.  This test won't work as-is in the 2.0.x codebase.  

I think I need to have a discussion with the pdfbox maintainers about how they want to handle test coverage for my proposed change.